### DISCUSSION TITLE
Will unchecking "Prevent change to network and firewall parameters" cause any leaks out of sandbox?

### DISCUSSION THREAD

**Discussion by MaryAtlas** [sandboxie-plus/Sandboxie#4289] [Q&A Feedback]
Hello. I require to add a few lines to firewall. An example line like this and similar:
for /R "C:\Program Files\Red Giant" %%X in (*.exe) do (
        echo Blocking: %%~nX
        netsh advfirewall firewall add rule name="%%~nX adobe-block" dir=out program="%%X" action=block
        netsh advfirewall firewall add rule name="%%~nX adobe-block" dir=in program="%%X" action=block
    )

With default settings, I am getting error that it could not communicate with Firewall service. I think the solution is to uncheck "Prevent change to network and firewall parameters"? However, the Firewall line is suggested to me to enable running a potential virus app. Will "Prevent change to network and firewall parameters" have any potential infection/damage to my system that is outside of sandbox?

**Comment by offhub** [comment]
I don't think the `BlockNetParam=n` setting will be sufficient/necessary for using NETSH. For NETSH, administrator privileges and certain IPC channels need to be open. You would be granting excessive permissions just to allow the application through the firewall.

Why don't you add the rules through the host? After all, even if you make the necessary settings in the sandbox to allow NETSH, permissions will still be set on the host.

**Reply by MaryAtlas** [reply]
By "host", do you mean my system outside of the sandbox? If I add them directly into Firewall, will there be potential breach into my system, like maybe accidentally opening a port of virus breaching my system, or something else?

**Reply by offhub** [reply]
> By "host", do you mean my system outside of the sandbox?

Yes.

> If I add them directly into Firewall will there be potential breach into my system, like maybe accidentally opening a port of virus breaching my system, or something else?

Even if the program is blocked by the firewall, I wouldn’t take the risk of running anything that could be potentially harmful, even in a sandbox environment.

**Reply by MaryAtlas** [reply]
Hmm? I thought that is what Sandboxie is for? To run any harmful software in a sandbox environment so to test it before running in the main system.

**Reply by offhub** [reply]
If you know a piece of software is harmful, it's best to avoid running it altogether. Prevention is key. While tools like Sandboxie are great for testing **untrusted** software, harmful software can sometimes detect when it’s in a sandbox environment and alter its behavior accordingly. It might withhold malicious actions or disguise its true nature to avoid detection. This makes it crucial to be cautious, even with sandboxing tools. If you're already aware of the risks, it's safer to steer clear entirely. Always prioritize your system's security and stick to trusted software!

**Reply by MaryAtlas** [reply]
Great piece of info on software sandbox detection. This also means that if I restrict the use of the harmful software inside sandbox, and never letting it outside, my system will be secured. Now whether or not messing with firewall will give it a chance to breakout, that is the million dollar question.

**Comment by love-code-yeyixiao** [comment]
If there is no any open service running in your system,then allowing won＇t make huge impact for just your system.






---Original---
From: ***@***.***&gt;
Date: Fri, Oct 11, 2024 20:54 PM
To: ***@***.***&gt;;
Cc: ***@***.***&gt;;
Subject: Re: [sandboxie-plus/Sandboxie] Will unchecking "Prevent change tonetwork and firewall parameters" cause any leaks out of sandbox? (Discussion#4289)




 
Great piece of info on software sandbox detection. This also means that if I restrict the use of the harmful software inside sandbox, and never letting it outside, my system will be secured. Now whether or not messing with firewall will give it a chance to breakout, that is the million dollar question.

**Reply by MaryAtlas** [reply]
What do you mean by "open service"? It would be great if you could explain more for me.
